tests/components/nws/test_weather.py

```diff
@@ -355,7 +322,7 @@ async def test_forecast_service(
 
     assert instance.update_observation.call_count == 2
     assert instance.update_forecast.call_count == 2
-    assert instance.update_forecast_hourly.call_count == 1
+    assert instance.update_forecast_hourly.call_count == 2
```

Comment:
Failure in this test on line above shows that `CoordinatorWeatherEntity.forecast_coordinators["twice_daily"]` is not called on a time basis, at least in this integration implementation.  This PR tries to use the `CoordinatorWeatherEntity` methods as much as possible here however.

Reply:
`CoordinatorWeatherEntity` does not have a `coordinator.async_add_listener` call in `async_added_to_hass`.

https://github.com/home-assistant/core/blob/a8025a8606fa63be570fb43f0f0643068bef8844/homeassistant/components/weather/__init__.py#L1116-L1129

It only occurs if a subscription is performed first:

def _async_subscription_started(
        self,
        forecast_type: Literal["daily", "hourly", "twice_daily"],
    ) -> None:
        """Start subscription to forecast_type."""
        if not (coordinator := self.forecast_coordinators[forecast_type]):
            return
        self.unsub_forecast[forecast_type] = coordinator.async_add_listener(
            partial(self._handle_forecast_update, forecast_type)
        )

Reply:
`coordinator.async_add_listener` is now called in `async_added_to_hass` to fix this.